src/sagemaker/model.py

```diff
@@ -2222,6 +2226,76 @@ def update_customer_metadata(self, customer_metadata_properties: Dict[str, str])
         sagemaker_session = self.sagemaker_session or sagemaker.Session()
         sagemaker_session.sagemaker_client.update_model_package(**update_metadata_args)
 
+    def update_inference_specification(
+        self,
+        containers: Dict = None,
+        image_uris: List[str] = None,
+        content_types: List[str] = None,
+        response_types: List[str] = None,
+        inference_instances: List[str] = None,
+        transform_instances: List[str] = None,
+    ):
+        """Inference specification to be set for the model package
+
+        Args:
+            containers (dict): The Amazon ECR registry path of the Docker image
+                that contains the inference code.
+            image_uris (List[str]): The ECR path where inference code is stored.
+            content_types (list[str]): The supported MIME types
+                for the input data.
+            response_types (list[str]): The supported MIME types
+                for the output data.
+            inference_instances (list[str]): A list of the instance
+                types that are used to generate inferences in real-time (default: None).
+            transform_instances (list[str]): A list of the instance
+                types on which a transformation job can be run or on which an endpoint can be
+                deployed (default: None).
+
+        """
+        sagemaker_session = self.sagemaker_session or sagemaker.Session()
+        if containers is not None and image_uris is not None:
+            raise ValueError("Cannot have both containers and image_uris.")
+        if containers is None and image_uris is None:
+            raise ValueError("Should have either containers or image_uris for inference.")
```

Comment:
you can have it as 
`(containers is not None) ^ (image_uris is None):`

raise ValueError("Should have either containers or image_uris for inference.")